In [ ]:
import pandas as pd
import numpy as np
import matplotlib.pyplot as plt
import seaborn as sns

In [ ]:
data = pd.read_csv("/content/train.csv")
display(data)

Id  MSSubClass MSZoning  LotFrontage  LotArea Street Alley LotShape  \
0        1          60       RL         65.0     8450   Pave   NaN      Reg   
1        2          20       RL         80.0     9600   Pave   NaN      Reg   
2        3          60       RL         68.0    11250   Pave   NaN      IR1   
3        4          70       RL         60.0     9550   Pave   NaN      IR1   
4        5          60       RL         84.0    14260   Pave   NaN      IR1   
...    ...         ...      ...          ...      ...    ...   ...      ...   
1455  1456          60       RL         62.0     7917   Pave   NaN      Reg   
1456  1457          20       RL         85.0    13175   Pave   NaN      Reg   
1457  1458          70       RL         66.0     9042   Pave   NaN      Reg   
1458  1459          20       RL         68.0     9717   Pave   NaN      Reg   
1459  1460          20       RL         75.0     9937   Pave   NaN      Reg   

     LandContour Utilities  ... PoolArea PoolQC  Fence MiscFeature MiscVal  \
0            Lvl    AllPub  ...        0    NaN    NaN         NaN       0   
1            Lvl    AllPub  ...        0    NaN    NaN         NaN       0   
2            Lvl    AllPub  ...        0    NaN    NaN         NaN       0   
3            Lvl    AllPub  ...        0    NaN    NaN         NaN       0   
4            Lvl    AllPub  ...        0    NaN    NaN         NaN       0   
...          ...       ...  ...      ...    ...    ...         ...     ...   
1455         Lvl    AllPub  ...        0    NaN    NaN         NaN       0   
1456         Lvl    AllPub  ...        0    NaN  MnPrv         NaN       0   
1457         Lvl    AllPub  ...        0    NaN  GdPrv        Shed    2500   
1458         Lvl    AllPub  ...        0    NaN    NaN         NaN       0   
1459         Lvl    AllPub  ...        0    NaN    NaN         NaN       0   

     MoSold YrSold  SaleType  SaleCondition  SalePrice  
0         2   2008        WD         Normal     208500  
1         5   2007        WD         Normal     181500  
2         9   2008        WD         Normal     223500  
3         2   2006        WD        Abnorml     140000  
4        12   2008        WD         Normal     250000  
...     ...    ...       ...            ...        ...  
1455      8   2007        WD         Normal     175000  
1456      2   2010        WD         Normal     210000  
1457      5   2010        WD         Normal     266500  
1458      4   2010        WD         Normal     142125  
1459      6   2008        WD         Normal     147500  

[1460 rows x 81 columns]

- SalePrice - harga jual properti dalam dolar. Ini adalah variabel target yang ingin Anda prediksi.
- MSSubClass: Kelas bangunan
- MSZoning: Klasifikasi zonasi umum
- LotFrontage: Kaki jalan linier yang terhubung ke properti
- LotArea: Ukuran lot dalam kaki persegi
- Street: Jenis akses jalan
- Alley: Jenis akses gang
- LotShape: Bentuk umum properti
- LandContour: Kerataan properti
- Utilites: Jenis utilitas yang tersedia
- LotConfig: Konfigurasi banyak
- LandSlope: Kemiringan properti
- Neighborhood: Lokasi fisik dalam batas kota Ames
- Condition 1: Dekat dengan jalan utama atau rel kereta api
- Condition 2: Dekat dengan jalan utama atau rel kereta api (jika ada yang kedua)
- BldgType: Tipe hunian
- HouseStyle: Gaya hunian
- OverallQual: Keseluruhan material dan kualitas hasil akhir
- OverallCond: Peringkat kondisi keseluruhan
- YearBuilt: Tanggal konstruksi asli
- YearRemodAdd: Tanggal perombakan
- RoofStyle: Jenis atap
- RoofMatl : Bahan atap
- Exterior1st: Penutup eksterior pada rumah
- Exterior2nd: Penutup eksterior pada rumah (jika lebih dari satu bahan)
- MasVnrType: Jenis veneer pasangan bata
- MasVnrArea: Area veneer batu dalam kaki persegi
- ExterQual: Kualitas material eksterior
- ExterCond: Kondisi material eksterior saat ini
- Pondasi: Jenis pondasi
- BsmtQual : Ketinggian basement
- BsmtCond : Kondisi umum basement
- BsmtExposure: Dinding basement walkout atau tingkat taman
- BsmtFinType1: Kualitas area basement jadi
- BsmtFinSF1: Tipe 1 kaki persegi jadi
- BsmtFinType2: Kualitas area jadi kedua (jika ada)
- BsmtFinSF2: Tipe 2 kaki persegi jadi
- BsmtUnfSF: Area basement seluas satu kaki persegi yang belum selesai
- TotalBsmtSF: Total kaki persegi luas basement
- Heating: Jenis pemanasan
- HeatingQC: Kualitas dan kondisi pemanasan
- CentralAir: AC sentral
- Elictrical: Sistem kelistrikan
- 1stFlrSF: Kaki persegi Lantai Pertama
- 2ndFlrSF: Lantai dua kaki persegi
- LowQualFinSF: Kaki persegi jadi berkualitas rendah (semua lantai)
- GrLivArea: Kaki persegi ruang tamu di atas permukaan tanah (tanah).
- BsmtFullBath: Kamar mandi lengkap di ruang bawah tanah
- BsmtHalfBath: Setengah kamar mandi bawah tanah
- FullBath: Kamar mandi lengkap di atas kelas
- HalfBath: Setengah kamar mandi di atas kelas
- Bedroom: Jumlah kamar tidur di atas lantai basement
- Kitchen: Jumlah dapur
- KitchenQual : Kualitas dapur
- TotRmsAbvGrd : Total kamar diatas kelas (tidak termasuk kamar mandi)
- Functional: Peringkat fungsionalitas rumah
- Fireplaces: Jumlah perapian
- FireplaceQu: Kualitas perapian
- GarageType: Lokasi garasi
- GarageYrBlt: Tahun garasi dibangun
- GarageFinish: Finishing interior garasi
- GarageCars: Ukuran garasi dalam kapasitas mobil
- GarageArea: Ukuran garasi dalam kaki persegi
 GarageQual: Kualitas garasi
- GarageCond: Kondisi garasi
- PavedDrive: Jalan masuk beraspal
- WoodDeckSF: Luas dek kayu dalam kaki persegi
- OpenPorchSF: Area teras terbuka dalam satuan kaki persegi
- EnclosedPorch: Area teras tertutup dalam ukuran kaki persegi
- 3SsnPorch: Area teras tiga musim dalam kaki persegi
- ScreenPorch: Area teras layar dalam kaki persegi
- PoolArea: Area kolam dalam kaki persegi
- PoolQC: Kualitas kolam
- Fence: Kualitas pagar
- MiscFeature: Fitur lain-lain yang tidak tercakup dalam kategori lain
- MiscVal: $Nilai fitur lain-lain
- MoSold: Bulan Terjual
- YrSold: Tahun Terjual
- SaleType: Jenis penjualan
- SaleCondition: Kondisi penjualan

## Missing Value

In [ ]:
data.info()

Total data yang seharusnya ada adalah 1460. Dalam data ini terdapat beberapa kolom yang datanya Missing. Diantaranya adalah :
- LotFrontage : 259 Data Missing Value
- Alley : 1369 Data Missing Value
- MasVnrType : 8 Data Missing Value
- MasVnrArea : 8 Data Missing Value
- BsmtQual : 37 Data Missing Value
- BsmtCond : 37 Data Missing Value
- BsmtExposure : 38 Data Missing Value
- BsmtFinType1 : 37 Data Missing Value
- BsmtFinType2 : 38 Data Missing Value
- Electrical : 1 Data Missing Value
- FireplaceQu : 690 Data Missing Value
- GarageType : 81 Data Missing Value
- GarageYrBlt : 81 Data Missing Value
- GarageFinish : 81 Data Missing Value
- GarageQual : 81 Data Missing Value
- GarageCond : 81 Data Missing Value
- PoolQC : 1453 Data Missing Value
- Fence : 1179 Data Missing Value
- MiscFeature : 1406 Data Missing Value

Dari sini saya membuat ketentuan yaitu jika data Missing Value mencapai lebih dari 1/4 jumlah data, maka saya akan menghapusnya. Karena sudah tidak merepresentasikan data. Jika datanya <= 1/4 jumlah data maka saya akan mengisinya menggunakan mean atau median.

In [ ]:
data["OverallCond"]

0       5
1       8
2       5
3       5
4       5
       ..
1455    5
1456    6
1457    9
1458    6
1459    6
Name: OverallCond, Length: 1460, dtype: int64

In [ ]:
data.describe()

In [ ]:
data.describe(include=object)

In [ ]:
meanLotFrontage = np.mean(data.LotFrontage)
print("LotFrontage :", meanLotFrontage)

Disini saya mengatasi Missing Value pada kolom LotFrontage yaitu dengan mengisi nya menggunakan median

In [ ]:
data["LotFrontage"] = data["LotFrontage"].fillna(data["LotFrontage"].median())
data["LotFrontage"]

Disini saya menghapus kolom Alley karena data yang Missing Value lebih dari 1/4 jumlah data

In [ ]:
data = data.drop('Alley', axis=1)

In [ ]:
display(data)

In [ ]:
kolom_MasVnrType = data['MasVnrType']
print(kolom_MasVnrType)

In [ ]:
import statistics as stats
stats.mode(data['MasVnrType'])

In [ ]:
mode_MasVnrType = data['MasVnrType'].mode()[0]  # Mendapatkan mode (nilai paling sering muncul)
data['MasVnrType'] = data['MasVnrType'].fillna(mode_MasVnrType)

Disini saya menggunakan Mode untuk mengisi data Missing Value pada kolom MasVnrType

In [ ]:
meanMasVnrArea = np.mean(data.MasVnrArea)
print("MasVnrArea :", meanMasVnrArea)

In [ ]:
mode_MasVnrArea = data['MasVnrArea'].mode()[0]  # Mendapatkan mode (nilai paling sering muncul)
data['MasVnrArea'] = data['MasVnrArea'].fillna(mode_MasVnrArea)

Disini saya menggunakan mode untuk mengisi Missing Value

In [ ]:
kolom_BsmtQual = data['BsmtQual']
print(kolom_BsmtQual)

In [ ]:
display(data)

In [ ]:
stats.mode(data['BsmtQual'])

In [ ]:
mode_BsmtQual = data['BsmtQual'].mode()[0]  # Mendapatkan mode (nilai paling sering muncul)
data['BsmtQual'] = data['BsmtQual'].fillna(mode_BsmtQual)

In [ ]:
data['BsmtCond']

In [ ]:
mode_BsmtCond = data['BsmtCond'].mode()[0]
data['BsmtCond'] = data['BsmtCond'].fillna(mode_BsmtCond)

In [ ]:
data['BsmtExposure']

NameError: name 'data' is not defined

In [ ]:
stats.mode(data['BsmtExposure'])

In [ ]:
mode_BsmtExposure = data['BsmtExposure'].mode()[0]
data['BsmtExposure'] = data['BsmtExposure'].fillna(mode_BsmtExposure)

In [ ]:
data['BsmtFinType1']

In [ ]:
stats.mode(data['BsmtFinType1'])

In [ ]:
mode_BsmtFinType1 = data['BsmtFinType1'].mode()[0]
data['BsmtFinType1'] = data['BsmtFinType1'].fillna(mode_BsmtFinType1)

In [ ]:
stats.mode(data['BsmtFinType2'])

In [ ]:
mode_BsmtFinType2 = data['BsmtFinType2'].mode()[0]
data['BsmtFinType2'] = data['BsmtFinType2'].fillna(mode_BsmtFinType2)

In [ ]:
data['Electrical']

In [ ]:
stats.mode(data['Electrical'])

In [ ]:
mode_Electrical = data['Electrical'].mode()[0]
data['Electrical'] = data['Electrical'].fillna(mode_Electrical)

In [ ]:
data.info()

In [ ]:
FireplaceQu_TA = data.loc[data['FireplaceQu'] == 'TA', ['FireplaceQu']]
FireplaceQu_TA

In [ ]:
FireplaceQu_Gd = data.loc[data['FireplaceQu'] == 'Gd', ['FireplaceQu']]
FireplaceQu_Gd

In [ ]:
FireplaceQu_Ex = data.loc[data['FireplaceQu'] == 'Ex', ['FireplaceQu']]
FireplaceQu_Ex

In [ ]:
stats.mode(data['FireplaceQu'])

## Feature

- Total luas bangunan : 1stFlrSF + 2ndFlrSF + TotalBsmtSF.
- Usia bangunan : YrBuilt - YrSold
- Jumlah kamar mandi : FullBath + HalfBath
- Total luas teras : WoodDeckSF + OpenPorchSF + EnclosedPorch + 3SsnPorch + ScreenPorch.
- Rating Kualitas rumah : (OverallQual + OverallCond)/2
- Total ruangan : Bedroom + TotRmsAbvGrd
- Kondisi Garasi : (GarageQual + GarageCond) /2
-
